### PR TITLE
chore: add internal method for utility context bindings

### PR DESCRIPTION
**Pull request by JoelEinbinder**
This only works in chromium, throws for other browsers. Split from #4519.

### CODE REVIEW

File: src/server/dom.ts [hunk 1]
```diff
@@ -28,11 +28,12 @@ import { FatalDOMError, RetargetableDOMError } from './common/domErrors';
 export class FrameExecutionContext extends js.ExecutionContext {
   readonly frame: frames.Frame;
   private _injectedScriptPromise?: Promise<js.JSHandle>;
-  private _debugScriptPromise?: Promise<js.JSHandle | undefined>;
+  readonly world: types.World|null;
```
Comment (dgozman) [inline src/server/dom.ts:31]: Shouldn't eslint complain about missing spaces around `|`?

File: src/server/firefox/ffPage.ts [hunk 2]
```diff
@@ -134,11 +134,14 @@ export class FFPage implements PageDelegate {
...
       return;
     const delegate = new FFExecutionContext(this._session, executionContextId);
-    const context = new dom.FrameExecutionContext(delegate, frame);
+    let worldName: types.World|null = null;
     if (auxData.name === UTILITY_WORLD_NAME)
-      frame._contextCreated('utility', context);
-    else if (!auxData.name)
-      frame._contextCreated('main', context);
+      worldName = 'utility';
+    else if (!auxData)
```
Comment (dgozman) [inline src/server/firefox/ffPage.ts:140]: This used to be `!auxData.name`. Did something change in the protocol?

File: src/server/browserContext.ts [hunk 3]
```diff
@@ -184,14 +184,15 @@ export abstract class BrowserContext extends EventEmitter {
   }
 
   async exposeBinding(name: string, needsHandle: boolean, playwrightBinding: frames.FunctionWithSource): Promise<void> {
+    const identifier = 'main:' + name;
```
Comment (dgozman) [inline src/server/browserContext.ts:187]: Let's make a static function `PageBinding.identifier(name, world)`?

File: src/server/chromium/crPage.ts [hunk 4]
```diff
@@ -683,17 +684,16 @@ class FrameSession {
...
+      this._client.send('Page.addScriptToEvaluateOnNewDocument', { source: binding.source, worldName })
     ]);
   }
 
   async _onBindingCalled(event: Protocol.Runtime.bindingCalledPayload) {
     const context = this._contextIdToContext.get(event.executionContextId)!;
-    const pageOrError = await this._crPage.pageOrError();
-    if (!(pageOrError instanceof Error))
-      this._page._onBindingCalled(event.payload, context);
+    await this._page._onBindingCalled(event.payload, context);
```
Comment (dgozman) [inline src/server/chromium/crPage.ts:696]: Why this change? I hoped we have a test for a binding in errored pages, but guess not.

File: src/server/dom.ts (hunk 1 above)
Comment (JoelEinbinder) [reply src/server/dom.ts:31]: Apparently there is no rule that covers this in eslint. [space-infix-ops](https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/space-infix-ops.md) looked promising, but I upgraded our typescript-eslint package and tried it and it didn't work.

File: src/server/firefox/ffPage.ts (hunk 2 above)
Comment (JoelEinbinder) [reply src/server/firefox/ffPage.ts:140]: No, I just messed up copy pasting.

File: src/server/chromium/crPage.ts (hunk 4 above)
Comment (JoelEinbinder) [reply src/server/chromium/crPage.ts:696]: I thought it was unnecessary. I'll put it back.

**Review by dgozman — APPROVED** [review]
